### PR TITLE
Add topics for Swagger and On-prem

### PR DESCRIPTION
**Pull request by zdrawku**

### CODE REVIEW

File: en/on-prem-prerequisites-and-installation.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+title: App Builder On-premise Prerequisites and Installation
+_description: Learn how to start using App Builder On-premises version and what are the prerequisites to install and run it.
+_keywords: App builder, On-premise, Infragistics
+---
+
+# App Builder On-premise Prerequisites and Installation
+
+## Prerequisites
+
+This section lists the prerequisites to install the On-Prem version of AppBuilder and is intended for System Administrators who configure operational parameters that maintains and supports Linux/Mac OS/Windows.
```
Comment (morbidiam) [inline en/on-prem-prerequisites-and-installation.md:11]: This section lists the prerequisites to install the On-Prem version of AppBuilder and is intended for System Administrators who configure operational parameters that maintain and support Linux/Mac OS/Windows.

Comment (morbidiam) [inline en/on-prem-prerequisites-and-installation.md:11]: This section lists the prerequisites to install the On-Prem version of App Builder and is intended for System Administrators who configure operational parameters that maintain and support Linux/Mac OS/Windows.

File: en/on-prem-prerequisites-and-installation.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+<img class="responsive-img" style="box-shadow: 5px -4px 13px 1px grey" src="./images/login-parameters.png" />
+<p style="text-align:center;">Login Parameters Dialog</p>
+
+### Install Docker
+
+Windows guide -> [docs.microsoft.com guide](https://docs.microsoft.com/en-us/virtualization/windowscontainers/quick-start/set-up-environment?tabs=Windows-10-and-11#tabpanel_1_Windows-10-and-11)
+
+## Installation
+This section assumes that you already have docker and a sql server database installed.
```
Comment (morbidiam) [inline en/on-prem-prerequisites-and-installation.md:37]: This section assumes that you already have docker and an SQL server database installed.

File: en/on-prem-prerequisites-and-installation.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+3 - Open a terminal or command prompt window at the extracted location.
+4 - Load and verify the image.
+
+Run:
+
+```
+docker load --input appbuilder.tar
+```
+
+In order to verify that the image is correctly loaded, example with the table below
```
Comment (morbidiam) [inline en/on-prem-prerequisites-and-installation.md:52]: I couldn't quite get this sentence.  Do you mean something like:

In order to verify that the image is correctly loaded, see the example with the table below.

File: en/on-prem-prerequisites-and-installation.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+5 - Run the container:
+
+```
+docker run --restart always -p 80:5000 -e "ConnectionStrings:AppBuilderConnection=Data Source=<your-sql-database-ip>,<your-sql-database-port>;Database=<your-sql-database-name>;User ID=<your-sql-database-user>;Password=<your-sql-database-password>;Connect Timeout=15;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False" -v <external-folder-for-logs>:/appbuilder/logs -v <external-folder-for-storage>:/appbuilder/storage --name appbuilder appbuilder:1.0
+```
+
+6 - Open your browser and type `http://localhost/`
+
+Note: or if you are using Docker Desktop, go to Containers/Apps, find your container and click `Open in browser`
```
Comment (morbidiam) [inline en/on-prem-prerequisites-and-installation.md:70]: Note: If you are using Docker Desktop, go to Containers/Apps, find your container and click `Open in browser`.

File: en/on-prem-prerequisites-and-installation.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+Note: or if you are using Docker Desktop, go to Containers/Apps, find your container and click `Open in browser` 
+
+
+<img class="responsive-img" style="box-shadow: 5px -4px 13px 1px grey" src="./images/docker-apps.png" />
+<p style="text-align:center;">Docker Containers/Apps</p>
+
+### Updates
+
+
+1 - Follow the first 4 steps of the first time installation  with the newly published zip file.
```
Comment (morbidiam) [inline en/on-prem-prerequisites-and-installation.md:79]: I think the full stop should be removed as the following sentences don't end with one.

File: en/on-prem-prerequisites-and-installation.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+docker stop appbuilder
+```
+
+4 - Remove the container
+
+```
+docker rm appbuilder
+```
+
+5 - Run the container with the same command as the one used in step 5. of the first time installation.
```
Comment (morbidiam) [inline en/on-prem-prerequisites-and-installation.md:105]: again, remove the full stop.

File: en/on-prem-prerequisites-and-installation.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+
+## Activation
+This section assumes that you already installed the On-Prem instance and it is now up and running.
+
+When the server is first started a prompt dialog will provide you with _Installation ID_ and a _Authentication key_ will be requested. Send this Installation ID to our [Sales department](https://www.infragistics.com/about-us/contact-us#sales) based on your region and we will provide you with _Authentication key_ to activate the server.
```
Comment (morbidiam) [inline en/on-prem-prerequisites-and-installation.md:111]: When the server is first started, a prompt dialog will provide you with _Installation ID_ and a _Authentication key_ will be requested. Send this Installation ID to our [Sales department](https://www.infragistics.com/about-us/contact-us#sales) based on your region and we will provide you with _Authentication key_ to activate the server.

File: en/on-prem-prerequisites-and-installation.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+
+<img class="responsive-img" style="box-shadow: 5px -4px 13px 1px grey" src="./images/activate-app-builder.png" />
+<p style="text-align:center;">Activate App Builder</p>
+
+> [!NOTE]
+> You will be warn through the UI thirty days before the expiration of your key.
```
Comment (morbidiam) [inline en/on-prem-prerequisites-and-installation.md:118]: Thirty days before the expiration of your key, you will receive a warning message directly through the UI.

OR VERSION 2

You will receive a warning message directly through the UI thirty days before your key expires.

File: en/on-prem-prerequisites-and-installation.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+
+## Troubleshooting
+### Docker Desktop on Windows
+[Docker Desktop on Windows is not starting automatically without Login to Windows machine](https://github.com/docker/for-win/issues/6670) - Docker team doesn't  recommend Docker Desktop for _production_ workloads. You should use Docker on a Linux box, or Docker for Windows Server if you want Windows containers.
```
Comment (morbidiam) [inline en/on-prem-prerequisites-and-installation.md:123]: [Docker Desktop on Windows doesn't start automatically without Login to Windows machine](https://github.com/docker/for-win/issues/6670) - Docker team doesn't  recommend Docker Desktop for _production_ workloads. You should use Docker on a Linux box or Docker for Windows Server if you want Windows containers.

File: en/using-data-in-your-app.md [hunk 10]
```diff
@@ -5,14 +5,33 @@ _keywords: App builder, Indigo Design, Infragistics, Data Sources, Data Binding
 ---
 
 # Data Feature overview 
-The App Builder Data features lets users add, edit and use external data source in the application they are developing. By default, the App Builder comes with a mock Northwind data source which users can use for reference. Currently, a JSON type data sources can be added, enabling users to upload their own JSON file or provide a Rest API URL with the data source. All added data sources are placed in the data toolbox and users can expand/collapse each data source in order to see the included tables and selected fields.
```
Comment (morbidiam) [inline en/using-data-in-your-app.md:8]: The App Builder Data features let users add, edit and use external data source in the application they are developing. By default, the App Builder comes with a mock Northwind data source which users can use for reference. Currently, JSON type data sources can be added, enabling users to upload their own JSON file or provide a Rest API URL with the data source. All added data sources are placed in the data toolbox and users can expand/collapse each data source in order to see the included tables and selected fields.

File: en/using-data-in-your-app.md [hunk 11]
```diff
@@ -5,14 +5,33 @@ _keywords: App builder, Indigo Design, Infragistics, Data Sources, Data Binding
...
+The App Builder Data features lets users add, edit and use external data source in the application they are developing. By default, the App Builder comes with a mock Northwind data source which users can use for reference. Below you can find different ways to set up Rest API data sources:
```
Comment (morbidiam) [inline en/using-data-in-your-app.md:8]: The App Builder Data features let users add, edit and use external data source in the application they are developing. By default, the App Builder comes with a mock Northwind data source which users can use for reference. Below you can find different ways to set up Rest API data sources:

File: en/using-data-in-your-app.md [hunk 12]
```diff
@@ -5,14 +5,33 @@ _keywords: App builder, Indigo Design, Infragistics, Data Sources, Data Binding
...
+
+**OpenAPI Specification** (formerly Swagger Specification) is an API description format for REST APIs. An OpenAPI file allows you to describe your entire API, including:
+
+- Available endpoints (/users) and operations on each endpoint (GET /users, POST /users)
+[ADD IMAGE OR GIF]
+- Operation parameters Input and output for each operation
+[ADD IMAGE OR GIF]
+- Authentication methods
+[ADD IMAGE OR GIF]
+- Contact information, license, terms of use and other information.
```
Comment (morbidiam) [inline en/using-data-in-your-app.md:31]: Remove the full point at the end of the sentence as the previous sentences do not end with one.

File: en/using-data-in-your-app.md [hunk 13]
```diff
@@ -5,14 +5,33 @@ _keywords: App builder, Indigo Design, Infragistics, Data Sources, Data Binding
...
+
+API specifications can be written in YAML or JSON. The format is easy to learn and readable to both humans and machines. The complete OpenAPI Specification can be found on GitHub: [OpenAPI 3.0 Specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md). More about OpenAPI [here](https://swagger.io/docs/specification/about/).
+[ADD IMAGE OR GIF]
 ## Adding a data source
 The App Builder Data feature enables users to add an external data sources, such as JSON (Rest API), Excel or CSV file, either by uploading a file, or linking to external publicly available source. Only Rest API option is available in the initial release of this feature.
```
Comment (morbidiam) [inline en/using-data-in-your-app.md:36]: The App Builder Data feature enables users to add external data sources, such as JSON (Rest API), Excel or CSV file, either by uploading a file, or linking to external publicly available source. Only Rest API option is available in the initial release of this feature.

**Review by morbidiam — COMMENTED** [review]
@zdrawku , I've completed my review. There were some punctuation errors and some minor edits that I made.

File: en/on-prem-prerequisites-and-installation.md (hunk 3 above)
Comment (zdrawku) [reply en/on-prem-prerequisites-and-installation.md:52]: Yes

**Pull request merged**